internal/protocoltest/restxml/serializers.go

```diff
@@ -0,0 +1,2354 @@
+// Code generated by smithy-go-codegen DO NOT EDIT.
+package restxml
+
+import (
+	"context"
+	"fmt"
+	smithy "github.com/awslabs/smithy-go"
+	"github.com/awslabs/smithy-go/httpbinding"
+	"github.com/awslabs/smithy-go/middleware"
+	smithytime "github.com/awslabs/smithy-go/time"
+	smithyhttp "github.com/awslabs/smithy-go/transport/http"
+)
+
+type awsRestxml_serializeOpXmlEnums struct {
+}
+
+func (*awsRestxml_serializeOpXmlEnums) ID() string {
+	return "awsRestxml_serializeOpXmlEnums"
+}
+
+func (m *awsRestxml_serializeOpXmlEnums) HandleSerialize(ctx context.Context, in middleware.SerializeInput, next middleware.SerializeHandler) (
+	out middleware.SerializeOutput, metadata middleware.Metadata, err error,
+) {
```

Comment:
General idea, what do you think about rendering these operation specific middleware per into a operation file? Either in the operation file it self, or a file specific to serde?   e.g. `api_op_PostContent.go`, or `api_op_PostContent_serde.go`  All operation serde middleware, and input/output helper functions could go in that one file.  

This would help breaking up the massive serialize and deserialize files.

Reply:
Middleware helpers for serde are generated in respective operation file. We can generate the operation to serde in respective file. It won't add much value apart from avoiding a giant serde file. This probably won't affect documentation generation as well (as the functions are unexported).